context-config-store/store/persistence/file_backed_config_persistence.go

```diff
@@ -0,0 +1,128 @@
+package persistence
+
+import (
+	"github.com/adrg/xdg"
+	api "github.com/kurtosis-tech/kurtosis/context-config-store/api/golang"
+	"github.com/kurtosis-tech/stacktrace"
+	"google.golang.org/protobuf/encoding/protojson"
+	"os"
+	"path"
+	"sync"
+)
+
+const (
+	applicationDirname = "kurtosis" // TODO: this is common to Kurtosis config, should refactor
+
+	fileName        = "context-config.json"
```

Comment:
can we just write Proto to disk? or do we want the user to be able to read the file

Reply:
I initially wrote the proto to disk straight, but I think writing JSON doesn't add a ton of complexity and is easier to look at (even though we don't expect anyone to look at this file, better safe than sorry)

Reply:
if our workflows don't involve people looking at the file on disk perhaps just keep it proto and the code becomes simpler?

YAGNI. No strong opinions though. Using proto for unions and then storing it as JSON seems complex. Maybe it can be simpler.

Reply:
I don't follow you on this... The only thing that changes is `proto.Marshal` becomes `protojson.Marshal` and same for `Unmarshal` (I removed the multiline thing). I mean, I'm happy to serialize the proto straight to disk, it's just that doing it in JSON doesn't add _any_ complexity to the code _at all_, and if at some point, for whatever reason, who knows, we want to look into this file, we keep this door open. Basically here cost = 0, value = 0 but mayyyybe value could be 0.5 who knows. So, really unless I'm missing something I don't understand your point here sorry

Reply:
Agree; the added complexity is very very little and compensates more than enough for the value it provides. 

Two different formats made me uneasy but it was a knee jerk.